### ISSUE TITLE
Add fallback behavior in tax task when plugins can not be installed

### ISSUE THREAD

**Issue by adrianduffell** [woothemes/woocommerce#51698]
**Prerequisites (mark completed items with an [x]):**
- [ ] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [ ] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

**Issue Description:**

Approximately 2-5% of sites are unable to install plugins for various reasons. We can detect some of these occurrences upfront and provide alternative means to complete the tax task.

## Acceptance criteria

* When user does not have `install_plugins` capability, display the message "Your user role does not have the capability to install plugins. Please ask an administrator to update your role or have an administrator complete this task. Learn more."
* When user does not have `manage_options` capability, display the message "Your user role does not have the capability to manage options. Please ask an administrator to update your role or have an administrator complete this task. Learn more."
* Link "Learn more" to `https://woocommerce.com/document/roles-capabilities/` 
* When WordPress.org is unavailable, enable taxes and redirect users to the tax settings task to update settings manually.

**Comment by rjchow** [comment]
To add on, I think we should also have a fallback for the WooCommerce payments task in the home screen tasklist that immediately fails on click as it cannot install the plugin.

This fallback should either be triggered when it is automatically detected that plugins cannot be installed for technical reasons, or after a user action causes a determination that the task cannot be completed with any modification of the user’s actions.

Additionally wrt the original topic, we could also somehow let the user report the failure out-of-band i.e not through the WP infrastructure but maybe github issues or similar

**Comment by adrianduffell** [comment]
>  I think we should also have a fallback for the WooCommerce payments task in the home screen tasklist that immediately fails on click as it cannot install the plugin

Yeah, in fact we may better to take a step back and put restrictions on the task list itself when when the user does not have permissions. To better understand use-cases, I've created an issue to add a tracks prop for the user's role here https://github.com/woocommerce/woocommerce/issues/52402

I'll label this as blocked until we have more info.

**Comment by timmy5685** [comment]
I've taken a look and marked this ready for the engineering team to pick up.

From what I can see - the blocker referenced in the previous comment has been completed so it looks like this is good to proceed with. I have marked it as ready for engineering.
